### ISSUE TITLE
ColorControl COLOR_MODE attribute not updated by color commands (TZ-644)

### ISSUE THREAD

**Issue by qcrist** [espressif/esp-zigbee-sdk#251]
### Answers checklist.

- [X] I have read the documentation [ESP Zigbee SDK Programming Guide](https://docs.espressif.com/projects/esp-zigbee-sdk) and the issue is not addressed there.
- [X] I have updated my IDF branch (master or release) and ESP Zigbee libs ([esp-zboss-lib](https://components.espressif.com/components/espressif/esp-zboss-lib) and [esp-zigbee-lib](https://components.espressif.com/components/espressif/esp-zigbee-lib)) to the latest version and checked that the issue is present there.
- [X] I have searched the issue tracker for a similar issue and not found a similar issue.

### IDF version.

v5.2

### esp-zigbee-lib version.

1.1.1

### esp-zboss-lib version.

1.1.1

### Espressif SoC revision.

ESP32-C6

### What is the expected behavior?

If a light device is in ColorMode=0x1(XY) and receives a MoveToColorTemperature request, the ColorMode attribute should be updated to 0x2.

### What is the actual behavior?

The device crashes

### Steps to reproduce.

1) Create a light cluster which supports XY and CT
```
    uint16_t color_capabilities = 0x18;
    uint16_t zero = 0;
    uint16_t mireds = 500;
    uint16_t mireds_max = ESP_ZB_ZCL_COLOR_CONTROL_COLOR_TEMP_PHYSICAL_MAX_MIREDS_DEFAULT_VALUE;
    uint16_t mireds_min = ESP_ZB_ZCL_COLOR_CONTROL_COLOR_TEMP_PHYSICAL_MIN_MIREDS_DEFAULT_VALUE;
    uint8_t color_mode = 1;

    esp_zb_attribute_list_t* esp_zb_color_cluster = esp_zb_zcl_attr_list_create(ESP_ZB_ZCL_CLUSTER_ID_COLOR_CONTROL);
    ESP_ERROR_CHECK(esp_zb_color_control_cluster_add_attr(esp_zb_color_cluster, ESP_ZB_ZCL_ATTR_COLOR_CONTROL_COLOR_CAPABILITIES_ID, &color_capabilities));
    ESP_ERROR_CHECK(esp_zb_color_control_cluster_add_attr(esp_zb_color_cluster, ESP_ZB_ZCL_ATTR_COLOR_CONTROL_CURRENT_X_ID, &zero));
    ESP_ERROR_CHECK(esp_zb_color_control_cluster_add_attr(esp_zb_color_cluster, ESP_ZB_ZCL_ATTR_COLOR_CONTROL_CURRENT_Y_ID, &zero));
    ESP_ERROR_CHECK(esp_zb_color_control_cluster_add_attr(esp_zb_color_cluster, ESP_ZB_ZCL_ATTR_COLOR_CONTROL_COLOR_TEMPERATURE_ID, &mireds));
    ESP_ERROR_CHECK(esp_zb_color_control_cluster_add_attr(esp_zb_color_cluster, ESP_ZB_ZCL_ATTR_COLOR_CONTROL_COLOR_MODE_ID, &color_mode));
    ESP_ERROR_CHECK(esp_zb_color_control_cluster_add_attr(esp_zb_color_cluster, ESP_ZB_ZCL_ATTR_COLOR_CONTROL_OPTIONS_ID, &zero));
    ESP_ERROR_CHECK(esp_zb_color_control_cluster_add_attr(esp_zb_color_cluster, ESP_ZB_ZCL_ATTR_COLOR_CONTROL_COLOR_TEMP_PHYSICAL_MIN_MIREDS_ID, &mireds_min));
    ESP_ERROR_CHECK(esp_zb_color_control_cluster_add_attr(esp_zb_color_cluster, ESP_ZB_ZCL_ATTR_COLOR_CONTROL_COLOR_TEMP_PHYSICAL_MAX_MIREDS_ID, &mireds_max));
    ESP_ERROR_CHECK(esp_zb_color_control_cluster_add_attr(esp_zb_color_cluster, ESP_ZB_ZCL_ATTR_COLOR_CONTROL_ENHANCED_COLOR_MODE_ID, &color_mode));

```
2) send a MoveToColorTemperature request
3) device crashes, as it is in XY color mode

### More Information.

Example full source
```
#include <esp_check.h>
#include <driver/gpio.h>
#include "nvs_flash.h"

#include "ha/esp_zigbee_ha_standard.h"

#if !defined CONFIG_ZB_ZCZR
#error Define ZB_ZCZR in idf.py menuconfig to compile light (Router) source code.
#endif

static void bdb_start_top_level_commissioning_cb(uint8_t mode_mask) {
    ESP_ERROR_CHECK(esp_zb_bdb_start_top_level_commissioning(mode_mask));
}

const char* TAG = "signal-handler";

void check_gpio8() {
    const gpio_config_t reset_cnf = {
            .pin_bit_mask = 1 << GPIO_NUM_9,
            .mode = GPIO_MODE_INPUT,
            .pull_up_en = GPIO_PULLUP_ENABLE,
            .pull_down_en = GPIO_PULLDOWN_DISABLE,
            .intr_type = GPIO_INTR_DISABLE,

    };
    gpio_config(&reset_cnf);

    const int read = gpio_get_level(GPIO_NUM_9);

    ESP_LOGI(TAG, "READ GPIO9: %d", read);
    if (read == 0 && !esp_zb_bdb_is_factory_new()) {
        esp_zb_factory_reset();
    }
}

[[maybe_unused]] void esp_zb_app_signal_handler(esp_zb_app_signal_t* signal_struct) {
    uint32_t* p_sg_p = signal_struct->p_app_signal;
    esp_err_t err_status = signal_struct->esp_err_status;
    auto sig_type = (esp_zb_app_signal_type_t) *p_sg_p;
    switch (sig_type) {
        case ESP_ZB_ZDO_SIGNAL_SKIP_STARTUP:
            ESP_LOGI(TAG, "Zigbee stack initialized");
            esp_zb_bdb_start_top_level_commissioning(ESP_ZB_BDB_MODE_INITIALIZATION);
            break;
        case ESP_ZB_BDB_SIGNAL_DEVICE_FIRST_START:
        case ESP_ZB_BDB_SIGNAL_DEVICE_REBOOT:
            if (err_status == ESP_OK) {
                ESP_LOGI(TAG, "Device started up in %s factory-reset mode", esp_zb_bdb_is_factory_new() ? "" : "non");
                if (esp_zb_bdb_is_factory_new()) {
                    ESP_LOGI(TAG, "Start network steering");
                    esp_zb_bdb_start_top_level_commissioning(ESP_ZB_BDB_MODE_NETWORK_STEERING);
                } else {
                    check_gpio8();
                    ESP_LOGI(TAG, "Device rebooted");
                }
            } else {
                ESP_LOGW(TAG, "Failed to initialize Zigbee stack (status: %s)", esp_err_to_name(err_status));
            }
            break;
        case ESP_ZB_BDB_SIGNAL_STEERING:
            if (err_status == ESP_OK) {
                esp_zb_ieee_addr_t extended_pan_id;
                esp_zb_get_extended_pan_id(extended_pan_id);
                ESP_LOGI(TAG,
                         "Joined network successfully (Extended PAN ID: %02x:%02x:%02x:%02x:%02x:%02x:%02x:%02x, PAN ID: 0x%04hx, Channel:%d, Short Address: 0x%04hx)",
                         extended_pan_id[7], extended_pan_id[6], extended_pan_id[5], extended_pan_id[4],
                         extended_pan_id[3], extended_pan_id[2], extended_pan_id[1], extended_pan_id[0],
                         esp_zb_get_pan_id(), esp_zb_get_current_channel(), esp_zb_get_short_address());
            } else {
                ESP_LOGI(TAG, "Network steering was not successful (status: %s)", esp_err_to_name(err_status));
                esp_zb_scheduler_alarm((esp_zb_callback_t) bdb_start_top_level_commissioning_cb,
                                       ESP_ZB_BDB_MODE_NETWORK_STEERING, 1000);
            }
            break;
        case ESP_ZB_NWK_SIGNAL_PERMIT_JOIN_STATUS:
            if (err_status == ESP_OK) {
                if (*(uint8_t*) esp_zb_app_signal_get_params(p_sg_p)) {
                    ESP_LOGI(TAG, "Network(0x%04hx) is open for %d seconds", esp_zb_get_pan_id(),
                             *(uint8_t*) esp_zb_app_signal_get_params(p_sg_p));
                } else {
                    ESP_LOGW(TAG, "Network(0x%04hx) closed, devices joining not allowed.", esp_zb_get_pan_id());
//                    esp_zb_factory_reset();
                }
            }
            break;
        case ESP_ZB_NLME_STATUS_INDICATION: {
            printf("%s, status: 0x%x\n", esp_zb_zdo_signal_to_string(sig_type), *(uint8_t*) esp_zb_app_signal_get_params(p_sg_p));
        }
            break;
        default:
            ESP_LOGI(TAG, "ZDO signal: %s (0x%x), status: %s", esp_zb_zdo_signal_to_string(sig_type), sig_type,
                     esp_err_to_name(err_status));
            break;
    }
}

static esp_err_t zb_action_handler(esp_zb_core_action_callback_id_t callback_id, const void* message) {
    ESP_LOGW(TAG, "Receive Zigbee action(0x%x) callback", callback_id);
    return ESP_OK;
}


extern "C" [[maybe_unused]] int app_main() {
    ESP_LOGW("MAIN", "app_main");
    esp_zb_platform_config_t config = {
            .radio_config = {
                    .radio_mode = RADIO_MODE_NATIVE,
            },
            .host_config = {
                    .host_connection_mode = HOST_CONNECTION_MODE_NONE,
            },
    };


    esp_zb_cfg_t zb_nwk_cfg = {
            .esp_zb_role = ESP_ZB_DEVICE_TYPE_ROUTER,
            .install_code_policy = false, // I have no clue what this is or does
            .nwk_cfg = {
                    .zczr_cfg = {
                            .max_children = 10
                    }
            }
    };

    ESP_ERROR_CHECK(nvs_flash_init());
    ESP_ERROR_CHECK(esp_zb_platform_config(&config));

    esp_zb_init(&zb_nwk_cfg);

    esp_zb_on_off_cluster_cfg_t on_off_cfg{
            .on_off = ESP_ZB_ZCL_ON_OFF_ON_OFF_DEFAULT_VALUE
    };
    esp_zb_level_cluster_cfg_t level_cfg{
            .current_level = 0x7f
    };
    esp_zb_basic_cluster_cfg_t basic_cfg{
            .zcl_version = 8,
            .power_source = 1
    };
    esp_zb_identify_cluster_cfg_t identity_cfg{
            .identify_time = 0
    };
    esp_zb_groups_cluster_cfg_t groups_cfg{
            .groups_name_support_id = 0
    };
    esp_zb_scenes_cluster_cfg_t scenes_cfg{
            0, 0, 0, false, 0
    };

    uint16_t color_capabilities = 0x18;
    uint16_t zero = 0;
    uint16_t mireds = 500;
    uint16_t mireds_max = ESP_ZB_ZCL_COLOR_CONTROL_COLOR_TEMP_PHYSICAL_MAX_MIREDS_DEFAULT_VALUE;
    uint16_t mireds_min = ESP_ZB_ZCL_COLOR_CONTROL_COLOR_TEMP_PHYSICAL_MIN_MIREDS_DEFAULT_VALUE;
    uint8_t color_mode = 2;

    esp_zb_attribute_list_t* esp_zb_color_cluster = esp_zb_zcl_attr_list_create(ESP_ZB_ZCL_CLUSTER_ID_COLOR_CONTROL);
    ESP_ERROR_CHECK(esp_zb_color_control_cluster_add_attr(esp_zb_color_cluster, ESP_ZB_ZCL_ATTR_COLOR_CONTROL_COLOR_CAPABILITIES_ID, &color_capabilities));
    ESP_ERROR_CHECK(esp_zb_color_control_cluster_add_attr(esp_zb_color_cluster, ESP_ZB_ZCL_ATTR_COLOR_CONTROL_CURRENT_X_ID, &zero));
    ESP_ERROR_CHECK(esp_zb_color_control_cluster_add_attr(esp_zb_color_cluster, ESP_ZB_ZCL_ATTR_COLOR_CONTROL_CURRENT_Y_ID, &zero));
    ESP_ERROR_CHECK(esp_zb_color_control_cluster_add_attr(esp_zb_color_cluster, ESP_ZB_ZCL_ATTR_COLOR_CONTROL_COLOR_TEMPERATURE_ID, &mireds));
    ESP_ERROR_CHECK(esp_zb_color_control_cluster_add_attr(esp_zb_color_cluster, ESP_ZB_ZCL_ATTR_COLOR_CONTROL_COLOR_MODE_ID, &color_mode));
    ESP_ERROR_CHECK(esp_zb_color_control_cluster_add_attr(esp_zb_color_cluster, ESP_ZB_ZCL_ATTR_COLOR_CONTROL_OPTIONS_ID, &zero));
    ESP_ERROR_CHECK(esp_zb_color_control_cluster_add_attr(esp_zb_color_cluster, ESP_ZB_ZCL_ATTR_COLOR_CONTROL_COLOR_TEMP_PHYSICAL_MIN_MIREDS_ID, &mireds_min));
    ESP_ERROR_CHECK(esp_zb_color_control_cluster_add_attr(esp_zb_color_cluster, ESP_ZB_ZCL_ATTR_COLOR_CONTROL_COLOR_TEMP_PHYSICAL_MAX_MIREDS_ID, &mireds_max));
    ESP_ERROR_CHECK(esp_zb_color_control_cluster_add_attr(esp_zb_color_cluster, ESP_ZB_ZCL_ATTR_COLOR_CONTROL_ENHANCED_COLOR_MODE_ID, &color_mode));


    esp_zb_cluster_list_t* c_list = esp_zb_zcl_cluster_list_create();
    ESP_ERROR_CHECK(esp_zb_cluster_list_add_basic_cluster(c_list, esp_zb_basic_cluster_create(&basic_cfg), ESP_ZB_ZCL_CLUSTER_SERVER_ROLE));
    ESP_ERROR_CHECK(esp_zb_cluster_list_add_identify_cluster(c_list, esp_zb_identify_cluster_create(&identity_cfg), ESP_ZB_ZCL_CLUSTER_SERVER_ROLE));
    ESP_ERROR_CHECK(esp_zb_cluster_list_add_groups_cluster(c_list, esp_zb_groups_cluster_create(&groups_cfg), ESP_ZB_ZCL_CLUSTER_SERVER_ROLE));
    ESP_ERROR_CHECK(esp_zb_cluster_list_add_scenes_cluster(c_list, esp_zb_scenes_cluster_create(&scenes_cfg), ESP_ZB_ZCL_CLUSTER_SERVER_ROLE));
    ESP_ERROR_CHECK(esp_zb_cluster_list_add_on_off_cluster(c_list, esp_zb_on_off_cluster_create(&on_off_cfg), ESP_ZB_ZCL_CLUSTER_SERVER_ROLE));
    ESP_ERROR_CHECK(esp_zb_cluster_list_add_level_cluster(c_list, esp_zb_level_cluster_create(&level_cfg), ESP_ZB_ZCL_CLUSTER_SERVER_ROLE));
    ESP_ERROR_CHECK(esp_zb_cluster_list_add_color_control_cluster(c_list, esp_zb_color_cluster, ESP_ZB_ZCL_CLUSTER_SERVER_ROLE));

    esp_zb_ep_list_t* esp_zb_ep_list = esp_zb_ep_list_create();
/* add created endpoint (cluster_list) to endpoint list */
    esp_zb_ep_list_add_ep(esp_zb_ep_list, c_list, 1, ESP_ZB_AF_HA_PROFILE_ID, ESP_ZB_HA_ON_OFF_LIGHT_DEVICE_ID);
    esp_zb_device_register(esp_zb_ep_list);
    esp_zb_core_action_handler_register(zb_action_handler);
    ESP_ERROR_CHECK(esp_zb_start(false));

    esp_zb_main_loop_iteration();
    return 0;
}

```

The opposite example with ColorMode set to 2, then a XY color change also crashes the device.

**Comment by xieqinan** [comment]
@qcrist ,

I have attached an example demonstrating this issue, where the feature works correctly. Could you please download and test it?
[HA_color_dimmable_example.zip](https://github.com/espressif/esp-zigbee-sdk/files/14459969/HA_color_dimmable_example.zip)

**Comment by qcrist** [comment]
Hmm, your example does not crash for me.

If I add the hue attribute to my previous example it also does not crash. 

Why does the hue attribute have to be defined for the device to properly handle the MoveToColorTemperature request?

**Comment by xieqinan** [comment]
@qcrist,

It is confirmed that there is a bug in the color mode change process within the esp-zigbee-sdk. We will address this issue in the next released version. You can utilize the workaround temporarily for your project.

**Comment by xieqinan** [comment]
Hello @qcrist 
This issue has been addressed in version 1.2.0. Could you please update and test it?

**Comment by qcrist** [comment]
Appears to be working now, thanks!
